src/Kiota.Builder/Writers/HTTP/CodeClassDeclarationWriter.cs

```diff
@@ -0,0 +1,346 @@
+                queryParameters,
+                pathParameters,
+                baseUrl
+            );
+
+            // Write the HTTP operation (e.g., GET, POST, PATCH, etc.)
+            writer.WriteLine($"{method.Name.ToUpperInvariant()} {url} HTTP/1.1");
+
+            // Write the request body if present
+            WriteRequestBody(method, writer);
+
+            // Write a separator if there are more items that follow
+            if (++currentIndex < methodCount)
+            {
+                writer.WriteLine();
+                writer.WriteLine("###");
+                writer.WriteLine();
+            }
+        }
+    }
+
+    /// <summary>
+    /// Retrieves all the HTTP methods of kind RequestExecutor for the given request builder class.
+    /// </summary>
+    /// <param name="requestBuilderClass">The request builder class containing the HTTP methods.</param>
+    /// <returns>A list of HTTP methods of kind RequestExecutor.</returns>
+    private static List<CodeMethod> GetHttpMethods(CodeClass requestBuilderClass)
+    {
+        return requestBuilderClass
+            .GetChildElements(true)
+            .OfType<CodeMethod>()
+            .Where(static element => element.IsOfKind(CodeMethodKind.RequestExecutor))
+            .ToList();
+    }
+
+    /// <summary>
+    /// Writes the request body for the given method to the writer.
+    /// </summary>
+    /// <param name="method">The method containing the request body.</param>
+    /// <param name="writer">The language writer to write the request body to.</param>
+    private static void WriteRequestBody(CodeMethod method, LanguageWriter writer)
+    {
+        // If there is a request body, write it
+        var requestBody = method.Parameters.FirstOrDefault(static param => param.IsOfKind(CodeParameterKind.RequestBody));
+        if (requestBody is null) return;
+
+        writer.WriteLine($"Content-Type: {method.RequestBodyContentType}");
+
+        // Empty line before body content
+        writer.WriteLine();
+
+        // Loop through the properties of the request body and write a JSON object
+        if (requestBody.Type is CodeType ct && ct.TypeDefinition is CodeClass requestBodyClass)
+        {
+            writer.StartBlock();
+            WriteProperties(requestBodyClass, writer);
+            writer.CloseBlock();
+        }
+    }
+
+    /// <summary>
+    /// Writes the properties of the given request body class to the writer.
+    /// </summary>
+    /// <param name="requestBodyClass">The request body class containing the properties.</param>
+    /// <param name="writer">The language writer to write the properties to.</param>
+    private static void WriteProperties(CodeClass requestBodyClass, LanguageWriter writer)
+    {
+        var properties = requestBodyClass.Properties
+            .Where(static prop => prop.IsOfKind(CodePropertyKind.Custom))
+            .ToArray();
+
+        var propertyCount = properties.Length;
+        var currentIndex = 0;
```

Comment:
After the refactoring, the `currentIndex` property is never incremented. So, we need to figure out if it's still needed.